docs/integrations/databases/firebase.mdx

```diff
@@ -5,163 +5,176 @@ description: Learn how to integrate Clerk into your Firebase application.
+To use Firebase auth, ensure authentication is enabled in your Firebase dashboard. To do so:
 
-<Steps>
+1. Navigate to your Firebase dashboard.
+1. In the sidebar navigation, expand **Build** and select **Authentication**.
+1. Enable authentication on this page, and enable any sign-in method you want, such as **Email/Password**.
 
-### Retrieve a Firebase user authentication token from Clerk
+### Add a Security Rule to your Firestore database
 
-The Firebase Web SDK requires an authentication token to sign in your users using Clerk as the custom authentication provider. This token can be retrieved calling the `getToken` method returned from the [`useAuth()`](/docs/references/react/use-auth) hook and calling it with the `integration_firebase` template parameter.
+To use Firestore with Clerk, ensure that you have defined [Security Rules](https://firebase.google.com/docs/firestore/security/get-started) that allow authenticated requests to access your database. For example:
 
-```jsx
-await getToken({ template: "integration_firebase" });
+```bash
+service cloud.firestore {
+  match /databases/{database}/documents {
+    match /{document=**} {
+      allow read, write: if request.auth != null;
+    }
+  }
+}
 ```
 
-The above method will return the token needed for the Firebase Web SDK to sign in your users with Clerk.
+### Get your Firebase config object
+
+To connect to your Firebase app in your code, you need a config object from your Firebase project. To find it:
+
+1. Visit [your Firebase project settings](https://console.firebase.google.com/project/_/settings/general/).
+1. Scroll down to the setup code snippet on this page, and copy the config. It should be similar to the following:
+    ```typescript
+    const firebaseConfig = {
+      apiKey: "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
+      authDomain: "clerk-example-xxxxx.firebaseapp.com",
+      databaseURL: "https://clerk-example-xxxxx-default-xxxx.firebaseio.com",
+      projectId: "clerk-test-xxxx",
+      storageBucket: "clerk-test-xxxx.appspot.com",
+      messagingSenderId: "012345678910",
+      appId: "1:012345678:web:abcdef123456hijklm",
+      measurementId: "G-ABC123DEF"
+    };
+    ```
+1. Save this information somewhere secure. You'll need it to connect to your Firebase app.
 
-### Sign in using the Firebase Web SDK
+See [Google's Firebase documentation](https://support.google.com/firebase/answer/7015592) for more information on the config object.
 
-<Callout type="info">
-  The Firebase Web SDK referenced is the browser installation of the official Firebase JavaScript SDK package on [npm](https://www.npmjs.com/package/firebase).
-</Callout>
+### Use Firebase with Clerk in your code
 
-To authenticate your users on Firebase using Clerk, you would need to call the `signInWithCustomToken` method of the Firebase Auth scope.
+The following example:
+- Uses Clerk to generate an authentication token for Firebase's API.
+- Creates a button for signing into your Firebase app.
```

Comment:
Pending the conversation we had - not sure how the auth thing works with firebase auth *and* clerk auth

```suggestion
- Creates a button for signing into your Firebase app using Firebase auth and Clerk auth.
```